tests/python/unittest/test_dynamic_shape.py

```diff
@@ -47,3 +47,26 @@ def hybrid_forward(self, F, data, index):
     assert_almost_equal(result.asnumpy(), result_nd)
     assert_almost_equal(data.grad.asnumpy(), data_grad_nd)
 
+def test_dynamic_shape_with_reshape():
+    # test dynamic shape op followed by reshape op
+    class _TestBlock(gluon.HybridBlock):
+
+        def __init__(self):
+            super(_TestBlock, self).__init__()
+
+        def hybrid_forward(self, F, data, index):
+            return F.contrib.boolean_mask(data, index).reshape((-1, ))
```

Comment:
let's also add test cases for deferred compute based implementation.

Reply:
Added a unit test for deferred compute mode

Reply:
@szha added more test cases for multiple hybridize calls